Classic.lp

```diff
@@ -0,0 +1,407 @@
+
+symbol ∧ᶜ p q ≔ ¬ ¬ p ∧ ¬ ¬ q; notation ∧ᶜ infix right 30;
+
+opaque symbol ∧ᶜᵢ [p q] : πᶜ p → πᶜ q → πᶜ (p ∧ᶜ q) ≔
+begin
+  assume p q Hp Hq;
+  assume Hnnp_and_Hnnq;
+  apply Hnnp_and_Hnnq;
+  apply ∧ᵢ
+  { assume Hnp; apply Hp;  refine Hnp }
+  { assume Hnq; apply Hq;  refine Hnq }
+end;
+
+opaque  symbol ∧ᶜₑ₁ [p q] : πᶜ (p ∧ᶜ q) → πᶜ p ≔
+begin
+  assume p q Hp_and_q;
+  assume Hnp;
+  apply Hp_and_q;
+  assume Hnnp_and_nnq;
+  refine (∧ₑ₁ Hnnp_and_nnq) Hnp;
+end;
+
+opaque  symbol ∧ᶜₑ₂ [p q] : πᶜ (p ∧ᶜ q) → πᶜ q ≔
+begin
+  assume p q Hp_and_q;
+  assume Hnq;
+  apply Hp_and_q;
+  assume Hnnp_and_nnq;
+  refine (∧ₑ₂ Hnnp_and_nnq) Hnq;
+end;
+
+symbol ⇒ᶜ p q ≔ ¬ ¬ p ⇒ ¬ ¬ q; notation ⇒ᶜ infix right 10;
+
+opaque symbol ⇒ᶜᵢ [p q]: (πᶜ p → πᶜ q) → πᶜ (p ⇒ᶜ q) ≔
+begin
+  assume p q Hpimpq;  
+  assume H;
+  apply H;
+  assume Hnnp Hnq;
+  apply (Hpimpq Hnnp Hnq)
+end;
+
+opaque symbol ⇒ᶜₑ [p q]: πᶜ (p ⇒ᶜ q) → πᶜ p → πᶜ q ≔
+begin
+  assume p q Hpimpq Hnnp Hq;  
+  apply Hpimpq;
+  assume H;
+  refine (H Hnnp Hq) 
+end;
+
+symbol ∀ᶜ [a] p ≔ `∀ x : τ a, ¬ ¬ (p x); notation ∀ᶜ quantifier;
+
+opaque symbol ∀ᶜᵢ p : (Π x, πᶜ (p x)) → πᶜ (∀ᶜ p) ≔
+begin
+  assume p Hnnpx Hnnnpx;
+  apply Hnnnpx;
+  assume x Hnnp;
+  apply Hnnpx x;
+  assume Hnp;
+  apply Hnnp;
+  apply Hnp
+end;
+
+opaque symbol ∀ᶜₑ [p] x : πᶜ (∀ᶜ p) → πᶜ (p x) ≔
+begin
+  assume p x Hnnforallnnp Hnnpx;
+  apply Hnnforallnnp;
+  assume Hforallnnp;
+  refine Hforallnnp x Hnnpx
+end;
+
+symbol ∃ᶜ [a] p ≔ `∃ x : τ a, ¬ ¬ (p x); notation ∃ᶜ quantifier;
+
+opaque symbol ∃ᶜᵢ [a] p (x: τ a) : πᶜ (p x) → πᶜ (∃ᶜ p) ≔
+begin
+  assume a p x Hpx;
+  assume Hnnnex;
+  apply Hnnnex;
+  apply ∃ᵢ
+  { refine x; }
+  { refine Hpx };
+end;
+
+opaque symbol ∃ᶜₑ [a] p : πᶜ (∃ᶜ p) → Π q, (Π x:τ a, πᶜ (p x) → πᶜ q) → πᶜ q ≔
+begin
+  assume a p Hexp q H;
+  assume Hnq;
+  apply Hexp;
+  assume H2;
+  print ∃ₑ;
+  apply ∃ₑ _ H2;
+  assume x Hnnp;
+  apply (H x Hnnp Hnq);
+end;
+
+symbol ∨ᶜ p q ≔ ¬ ¬ p ∨ ¬ ¬ q; notation ∨ᶜ infix right 20;
+
+opaque symbol ∨ᶜᵢ₁ [p q] : πᶜ p → πᶜ (p ∨ᶜ q) ≔
+begin
+  assume p q Hnnp;
+  assume Hnnp_or_nnq;
+  apply Hnnp_or_nnq;
+  apply ∨ᵢ₁;
+  refine Hnnp;
+end;
+
+opaque symbol ∨ᶜᵢ₂ [p q] : πᶜ q → πᶜ (p ∨ᶜ q) ≔
+begin
+  assume p q Hnnp;
+  assume Hnnp_or_nnq;
+  apply Hnnp_or_nnq;
+  apply ∨ᵢ₂;
+  refine Hnnp;
+end;
+
+opaque symbol ∨ᶜₑ [p q r] : πᶜ (p ∨ᶜ q) → (πᶜ p → πᶜ r) → (πᶜ q → πᶜ r) → πᶜ r ≔
+begin
+  assume p q r Hnnp_or_nnq Hnnp→nr Hnnq→nr Hr→⊥;
+  apply Hnnp_or_nnq;
+  assume H;
+  apply ∨ₑ H 
+  { assume Hnnp; apply Hnnp→nr Hnnp Hr→⊥ }
+  { assume Hnnq; apply Hnnq→nr Hnnq Hr→⊥ }
+end;
+
+opaque symbol ¬ᶜᵢ [p] : (πᶜ p → πᶜ ⊥) → πᶜ (¬ p) ≔
+begin
+  assume p Hnp;
+  assume Hnnp;
+  apply Hnnp;
+  assume Hp;
+  apply Hnp Hnnp;
+  assume Hbot;
+  refine Hbot;
+end;
+
+opaque symbol ¬ᶜₑ [p q] : πᶜ (¬ p) → πᶜ p → πᶜ q ≔
+begin
+  assume p q Hnnnp Hnnp Hnq;
+  refine Hnnnp Hnnp 
+end;
+
+constant symbol classic [p] : πᶜ (p ∨ᶜ ¬ p);
```

Comment:
Why is it an axiom and not a proved theorem?

Reply:
I added the proof. I just forgot to do it because I got my inspiration for this file from Coq Classic.v in Stdlib and it is an Axiom there.